include/crm/common/internal.h

```diff
@@ -188,6 +190,88 @@ char *pcmk__transition_key(int transition_id, int action_id, int target_rc,
 void pcmk__filter_op_for_digest(xmlNode *param_set);
 
 
+// bitwise arithmetic utilities
+
+/*!
+ * \internal
+ * \brief Set specified flags in a flag group
+ *
+ * \param[in] function    Function name of caller
+ * \param[in] line        Line number of caller
+ * \param[in] log_level   Log a message at this level
+ * \param[in] flag_type   Label describing this flag group (for logging)
+ * \param[in] target      Name of object whose flags these are (for logging)
+ * \param[in] flag_group  Flag group being manipulated
+ * \param[in] flags       Which flags in the group should be set
+ * \param[in] flags_str   Readable equivalent of \p flags (for logging)
+ *
+ * \return Possibly modified flag group
+ */
+static inline uint64_t
+pcmk__set_flags_as(const char *function, int line, uint8_t log_level,
+                   const char *flag_type, const char *target,
+                   uint64_t flag_group, uint64_t flags, const char *flags_str)
+{
+    uint64_t result = flag_group | flags;
+
+    if (result != flag_group) {
+        do_crm_log_unlikely(log_level,
+                            "%s flags 0x%.8llx (%s) for %s set by %s:%d",
+                            ((flag_type == NULL)? "Group of" : flag_type),
+                            (unsigned long long) flags,
```

Comment:
Would there be any portability benefit to using the PRIx64 macro in the format string, instead of casting to ULL?

Reply:
I think either approach is portable (any integer should be promotable to unsigned long long), but it does make sense to take advantage of inttypes.h for its intended purpose.

We currently define our own aliases for certain inttypes.h constants using configure.ac, for historical reasons. I think we should just use the inttypes.h constants directly. This PR's big enough I'd rather do that in a separate one, so I'll leave this alone for now :)

Reply:
Very true. I shouldn't have used the word "portability" there. We can always cast up.

And ack :)